Heapster v1.5.0-beta.0 panics regularly (nil pointer?)

It's currently restarted ~850 times over 5 days of runtime.

```
W1009 10:22:25.000654       1 manager.go:147] Failed to get all responses in time (got 7/12)
W1009 10:22:39.845271       1 manager.go:102] Failed to get kubelet_summary:10.33.244.3:10255 response in time
W1009 10:22:40.392933       1 manager.go:102] Failed to get kubelet_summary:10.33.244.254:10255 response in time
W1009 10:22:40.472368       1 manager.go:102] Failed to get kubelet_summary:10.33.244.3:10255 response in time
W1009 10:22:40.518392       1 manager.go:102] Failed to get kubelet_summary:10.33.244.3:10255 response in time
W1009 10:22:45.217932       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{weave 2017-10-07 11:07:16 +0000 UTC 0xc420ff94d0 0xc42047a2d0 0xc420ff9500 0xc420ff9560 []}
W1009 10:22:45.218313       1 manager.go:102] Failed to get kubelet_summary:10.33.244.28:10255 response in time
W1009 10:22:47.961319       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{weave 2017-10-07 11:07:16 +0000 UTC 0xc42193b590 0xc420e219f0 0xc42193b5c0 0xc42193b620 []}
W1009 10:22:47.961678       1 manager.go:102] Failed to get kubelet_summary:10.33.244.28:10255 response in time
W1009 10:22:49.654797       1 manager.go:102] Failed to get kubelet_summary:10.33.244.28:10255 response in time
W1009 10:22:57.367613       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{certificate-installation 2017-10-04 15:53:46 +0000 UTC 0xc4222e0de0 0xc421f312c0 0xc4222e0e10 0xc4222e0e70 []}
W1009 10:22:57.367840       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{ifzpostgres-ambient 2017-10-04 23:00:24 +0000 UTC 0xc4222e1920 0xc421f31720 0xc4222e1950 0xc4222e19b0 []}
W1009 10:22:57.367969       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{kafka-server 2017-10-04 15:53:46 +0000 UTC 0xc4221444e0 0xc421f31cc0 0xc422144510 0xc422144570 []}
W1009 10:22:57.368129       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{statsd 2017-10-04 22:56:59 +0000 UTC 0xc422145230 0xc420f0e0f0 0xc422145260 0xc4221452c0 []}
W1009 10:22:57.368155       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{django 2017-10-04 22:55:50 +0000 UTC 0xc422145500 0xc420f0e1e0 0xc422145530 0xc422145590 []}
W1009 10:22:57.368197       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{statsd-exporter 2017-10-04 22:57:31 +0000 UTC 0xc4221455f0 0xc420f0e230 0xc422145620 0xc422145680 []}
W1009 10:22:57.368222       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{testapp 2017-10-04 22:53:17 +0000 UTC 0xc422145830 0xc420f0e2d0 0xc422145860 0xc4221458c0 []}
W1009 10:22:57.368283       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{nginx 2017-10-04 15:53:47 +0000 UTC 0xc422145e90 0xc420f0e4b0 0xc422145ec0 0xc422145f20 []}
W1009 10:22:57.368385       1 summary.go:216] Two identical containers are reported and the older one is not terminated: &{prometheus-config-reloader 2017-10-04 15:53:57 +0000 UTC 0xc42241a4b0 0xc420f0e640 0xc42241a4e0 0xc42241a570 []}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1591838]

goroutine 357 [running]:
k8s.io/heapster/metrics/sources/summary.containerIsTerminated(0xc42230c840, 0xed1675f30, 0xc400000000, 0x25816e0, 0xc4203f66a0)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:213 +0x48
k8s.io/heapster/metrics/sources/summary.(*summaryMetricsSource).decodePodStats(0xc421dedc70, 0xc42259bcb0, 0xc42259bce0, 0xc421903be8)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:203 +0xaf1
k8s.io/heapster/metrics/sources/summary.(*summaryMetricsSource).decodeSummary(0xc421dedc70, 0xc421cc4500, 0x0)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:131 +0x336
k8s.io/heapster/metrics/sources/summary.(*summaryMetricsSource).ScrapeMetrics(0xc421dedc70, 0xed16d448c, 0xc400000000, 0x25816e0, 0xed16d44c8, 0xc400000000, 0x25816e0, 0x197511f)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:101 +0x262
k8s.io/heapster/metrics/sources.scrape(0x23f6f00, 0xc421dedc70, 0xed16d448c, 0xc400000000, 0x25816e0, 0xed16d44c8, 0xc400000000, 0x25816e0, 0x0)
	/go/src/k8s.io/heapster/metrics/sources/manager.go:169 +0x2a0
k8s.io/heapster/metrics/sources.(*sourceManager).ScrapeMetrics.func1(0xc4213a0e80, 0x23f6f00, 0xc421dedc70, 0xc421c6f020, 0xed16d448c, 0x0, 0x25816e0, 0xed16d44c8, 0x0, 0x25816e0, ...)
	/go/src/k8s.io/heapster/metrics/sources/manager.go:99 +0x155
created by k8s.io/heapster/metrics/sources.(*sourceManager).ScrapeMetrics
	/go/src/k8s.io/heapster/metrics/sources/manager.go:115 +0x387
```

Bumping the log level up to 8:
```
I1009 11:28:05.216627       1 round_trippers.go:420] Response Status: 200 OK in 78 milliseconds
I1009 11:28:05.216651       1 round_trippers.go:423] Response Headers:
I1009 11:28:05.216656       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:05.216659       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:05 GMT
I1009 11:28:05.222644       1 round_trippers.go:420] Response Status: 200 OK in 100 milliseconds
I1009 11:28:05.222661       1 round_trippers.go:423] Response Headers:
I1009 11:28:05.222665       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:05.222667       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:05 GMT
I1009 11:28:05.225100       1 round_trippers.go:420] Response Status: 200 OK in 146 milliseconds
I1009 11:28:05.225108       1 round_trippers.go:423] Response Headers:
I1009 11:28:05.225111       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:05.225114       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:05 GMT
I1009 11:28:05.229749       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-system/pod:weave-net-pq7f1/container:weave. Create time of containers are 2017-10-05 03:13:21 +0000 UTC and 2017-10-05 03:18:24 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:05.254827       1 round_trippers.go:420] Response Status: 200 OK in 113 milliseconds
I1009 11:28:05.254850       1 round_trippers.go:423] Response Headers:
I1009 11:28:05.254854       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:05.254858       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:05 GMT
I1009 11:28:06.243604       1 round_trippers.go:420] Response Status: 200 OK in 1163 milliseconds
I1009 11:28:06.243631       1 round_trippers.go:423] Response Headers:
I1009 11:28:06.243635       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:06 GMT
I1009 11:28:06.243638       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:06.363634       1 round_trippers.go:420] Response Status: 200 OK in 1280 milliseconds
I1009 11:28:06.363657       1 round_trippers.go:423] Response Headers:
I1009 11:28:06.363661       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:06.363664       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:06 GMT
I1009 11:28:07.273312       1 round_trippers.go:420] Response Status: 200 OK in 2064 milliseconds
I1009 11:28:07.273339       1 round_trippers.go:423] Response Headers:
I1009 11:28:07.273344       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:07 GMT
I1009 11:28:07.273347       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:07.276144       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-monitoring/pod:weave-scope-agent-826w5/container:agent. Create time of containers are 2017-10-01 01:43:20 +0000 UTC and 2017-10-01 01:41:55 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.284488       1 round_trippers.go:420] Response Status: 200 OK in 18129 milliseconds
I1009 11:28:23.284570       1 round_trippers.go:423] Response Headers:
I1009 11:28:23.284575       1 round_trippers.go:426]     Content-Type: application/json
I1009 11:28:23.284578       1 round_trippers.go:426]     Date: Mon, 09 Oct 2017 11:28:22 GMT
I1009 11:28:23.328078       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-extra/pod:container-log-ingest-bf032/container:ingest. Create time of containers are 2017-10-04 15:53:48 +0000 UTC and 2017-10-09 09:34:52 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328150       1 summary.go:202] Metrics reported from two containers with the same key: namespace:eventrouterkafka/pod:zookeeper-server-2/container:zookeeper-server. Create time of containers are 2017-10-05 00:59:50 +0000 UTC and 2017-10-04 15:53:46 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328189       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-extra/pod:registry-mirror-gcr-d5pnn/container:mirror-hostess. Create time of containers are 2017-10-09 11:02:08 +0000 UTC and 2017-10-05 00:35:19 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328202       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-extra/pod:registry-mirror-gcr-d5pnn/container:certificate-installation. Create time of containers are 2017-10-05 01:00:06 +0000 UTC and 2017-10-04 15:53:36 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328234       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-system/pod:kube-apiserver-1890097792-kkhm8/container:kube-apiserver. Create time of containers are 2017-10-05 00:59:50 +0000 UTC and 2017-10-04 15:53:06 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328269       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-monitoring/pod:weave-scope-agent-wbvxr/container:agent. Create time of containers are 2017-10-04 15:53:58 +0000 UTC and 2017-10-04 15:52:44 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328283       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-monitoring/pod:weave-scope-agent-wbvxr/container:agent. Create time of containers are 2017-09-29 23:38:01 +0000 UTC and 2017-10-04 15:53:58 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328291       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-monitoring/pod:weave-scope-agent-wbvxr/container:agent. Create time of containers are 2017-10-05 01:01:01 +0000 UTC and 2017-10-04 15:53:58 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328301       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-monitoring/pod:weave-scope-agent-wbvxr/container:agent. Create time of containers are 2017-09-30 19:23:05 +0000 UTC and 2017-10-05 01:01:01 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328308       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-monitoring/pod:weave-scope-agent-wbvxr/container:agent. Create time of containers are 2017-10-02 05:08:25 +0000 UTC and 2017-10-05 01:01:01 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.328332       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-extra/pod:registry-mirror-quay-jzlh9/container:certificate-installation. Create time of containers are 2017-10-05 01:00:06 +0000 UTC and 2017-10-04 15:53:36 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412037       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-extra/pod:registry-mirror-quay-jzlh9/container:mirror-hostess. Create time of containers are 2017-10-05 00:04:53 +0000 UTC and 2017-10-09 11:24:58 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412129       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-extra/pod:default-http-backend-1833742259-k010s/container:default-http-backend. Create time of containers are 2017-10-05 10:32:16 +0000 UTC and 2017-10-04 15:53:46 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412159       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-system/pod:pod-checkpointer-kubernetes-cit-kubernetes-cr0-2-1506506680/container:checkpoint. Create time of containers are 2017-10-05 00:59:50 +0000 UTC and 2017-10-04 15:53:46 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412183       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-e2etests-http-update/pod:e2etestapp-3929767995-5k1rb/container:testapp. Create time of containers are 2017-10-05 01:01:01 +0000 UTC and 2017-10-04 22:53:29 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412216       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-e2etests-deployment-service/pod:e2etestapp-395387445-h5ntp/container:testapp. Create time of containers are 2017-10-05 01:01:01 +0000 UTC and 2017-10-04 22:53:17 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412253       1 summary.go:202] Metrics reported from two containers with the same key: namespace:kube-system/pod:kube-controller-manager-3400949544-fmrrt/container:kube-controller-manager. Create time of containers are 2017-10-04 22:59:51 +0000 UTC and 2017-10-05 01:00:06 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412271       1 summary.go:202] Metrics reported from two containers with the same key: namespace:eventrouterkafka/pod:kafka-server-2/container:kafka-server. Create time of containers are 2017-10-05 05:02:41 +0000 UTC and 2017-10-05 11:54:47 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412282       1 summary.go:202] Metrics reported from two containers with the same key: namespace:eventrouterkafka/pod:kafka-server-2/container:kafka-server. Create time of containers are 2017-10-05 07:42:47 +0000 UTC and 2017-10-05 11:54:47 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412289       1 summary.go:202] Metrics reported from two containers with the same key: namespace:eventrouterkafka/pod:kafka-server-2/container:kafka-server. Create time of containers are 2017-10-05 01:53:27 +0000 UTC and 2017-10-05 11:54:47 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412295       1 summary.go:202] Metrics reported from two containers with the same key: namespace:eventrouterkafka/pod:kafka-server-2/container:kafka-server. Create time of containers are 2017-10-05 05:46:23 +0000 UTC and 2017-10-05 11:54:47 +0000 UTC. Metrics from the older container are going to be dropped.
I1009 11:28:23.412303       1 summary.go:202] Metrics reported from two containers with the same key: namespace:eventrouterkafka/pod:kafka-server-2/container:kafka-server. Create time of containers are 0001-01-01 00:00:00 +0000 UTC and 2017-10-05 11:54:47 +0000 UTC. Metrics from the older container are going to be dropped.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1591838]

goroutine 257 [running]:
k8s.io/heapster/metrics/sources/summary.containerIsTerminated(0xc42104f2c0, 0xed1681487, 0xc400000000, 0x25816e0, 0xc42102d338)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:213 +0x48
k8s.io/heapster/metrics/sources/summary.(*summaryMetricsSource).decodePodStats(0xc421c0c000, 0xc42164bbf0, 0xc42164bc20, 0xc421d79be8)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:203 +0xaf1
k8s.io/heapster/metrics/sources/summary.(*summaryMetricsSource).decodeSummary(0xc421c0c000, 0xc4211a9f80, 0x0)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:131 +0x336
k8s.io/heapster/metrics/sources/summary.(*summaryMetricsSource).ScrapeMetrics(0xc421c0c000, 0xed16d5404, 0xc400000000, 0x25816e0, 0xed16d5440, 0xc400000000, 0x25816e0, 0xa)
	/go/src/k8s.io/heapster/metrics/sources/summary/summary.go:101 +0x262
k8s.io/heapster/metrics/sources.scrape(0x23f6f00, 0xc421c0c000, 0xed16d5404, 0xc400000000, 0x25816e0, 0xed16d5440, 0x0, 0x25816e0, 0x0)
	/go/src/k8s.io/heapster/metrics/sources/manager.go:169 +0x2a0
k8s.io/heapster/metrics/sources.(*sourceManager).ScrapeMetrics.func1(0xc42108b130, 0x23f6f00, 0xc421c0c000, 0xc421b04a20, 0xed16d5404, 0x0, 0x25816e0, 0xed16d5440, 0x0, 0x25816e0, ...)
	/go/src/k8s.io/heapster/metrics/sources/manager.go:99 +0x155
created by k8s.io/heapster/metrics/sources.(*sourceManager).ScrapeMetrics
	/go/src/k8s.io/heapster/metrics/sources/manager.go:115 +0x387
```

--- a/metrics/sources/summary/summary.go
+++ b/metrics/sources/summary/summary.go
@@ -196,25 +196,15 @@ func (this *summaryMetricsSource) decodePodStats(metrics map[string]*MetricSet, 
 		// This check ensures that we are not replacing metrics of running container with metrics of terminated one if
 		// there are two exactly same containers reported by kubelet.
 		if _, exist := metrics[key]; exist {
-			glog.V(8).Infof("Metrics reported from two containers with the same key: %v. Create time of "+
+			glog.V(2).Infof("Metrics reported from two containers with the same key: %v. Create time of "+
 				"containers are %v and %v. Metrics from the older container are going to be dropped.", key,
 				container.StartTime.Time, metrics[key].CreateTime)
-			if containerIsTerminated(&container, metrics[key].CreateTime) {
+			if container.StartTime.Time.Before(metrics[key].CreateTime) {
 				continue
 			}
 		}
 		metrics[key] = this.decodeContainerStats(podMetrics.Labels, &container, false)
 	}
-}
-
-func containerIsTerminated(container *stats.ContainerStats, otherStartTime time.Time) bool {
-	if container.StartTime.Time.Before(otherStartTime) {
-		if *container.CPU.UsageNanoCores == 0 && *container.Memory.RSSBytes == 0 {
-			return true
-		}
-		glog.Warningf("Two identical containers are reported and the older one is not terminated: %v", container)
-	}
-	return false
 }
 
 func (this *summaryMetricsSource) decodeContainerStats(podLabels map[string]string, container *stats.ContainerStats, isSystemContainer bool) *MetricSet {

--- a/metrics/sources/summary/summary_test.go
+++ b/metrics/sources/summary/summary_test.go
@@ -63,6 +63,8 @@ const (
 	seedPod2Container0 = 6000
 	seedPod2Container1 = 7000
 	seedPod3Container0 = 9000
+	seedPod4           = 10000
+	seedPod4Container0 = 11000
 )
 
 const (
@@ -73,6 +75,7 @@ const (
 	pName1 = "pod1"
 	pName2 = "pod0" // ensure pName2 conflicts with pName0, but is in a different namespace
 	pName3 = "pod2"
+	pName4 = "pod4" // Regression test for #1838
 
 	cName00 = "c0"
 	cName01 = "c1"
@@ -80,6 +83,9 @@ const (
 	cName20 = "c1"      // ensure cName20 conflicts with cName01, but is in a different pod + namespace
 	cName21 = "runtime" // ensure that runtime containers are not renamed
 	cName30 = "c3"
+	cName40 = "c4" // Running, with cpu / memory stats
+	cName41 = "c4" // Terminated, has no CPU / Memory stats
+	cName42 = "c4" // Terminated, has blank CPU / Memory stats
 )
 
 var (
@@ -194,6 +200,18 @@ func TestDecodeSummaryMetrics(t *testing.T) {
 				},
 			},
 			},
+		}, {
+			PodRef: stats.PodReference{
+				Name:      pName4,
+				Namespace: namespace0,
+			},
+			StartTime: metav1.NewTime(startTime),
+			Network:   genTestSummaryNetwork(seedPod4),
+			Containers: []stats.ContainerStats{
+				genTestSummaryContainer(cName40, seedPod4Container0),
+				genTestSummaryTerminatedContainerNoStats(cName41),
+				genTestSummaryTerminatedContainerBlankStats(cName41),
+			},
 		}},
 	}
 
@@ -249,6 +267,11 @@ func TestDecodeSummaryMetrics(t *testing.T) {
 		seed:    seedPod2,
 		network: true,
 	}, {
+		key:     core.PodKey(namespace0, pName4),
+		setType: core.MetricSetTypePod,
+		seed:    seedPod4,
+		network: true,
+	}, {
 		key:     core.PodContainerKey(namespace0, pName0, cName00),
 		setType: core.MetricSetTypePodContainer,
 		seed:    seedPod0Container0,
@@ -287,6 +310,13 @@ func TestDecodeSummaryMetrics(t *testing.T) {
 		key:     core.PodContainerKey(namespace0, pName3, cName30),
 		setType: core.MetricSetTypePodContainer,
 		seed:    seedPod3Container0,
+		cpu:     true,
+		memory:  true,
+		fs:      containerFs,
+	}, {
+		key:     core.PodContainerKey(namespace0, pName4, cName40),
+		setType: core.MetricSetTypePodContainer,
+		seed:    seedPod4Container0,
 		cpu:     true,
 		memory:  true,
 		fs:      containerFs,
@@ -355,6 +385,22 @@ func genTestSummaryTerminatedContainer(name string, seed int) stats.ContainerSta
 	}
 }
 
+func genTestSummaryTerminatedContainerNoStats(name string) stats.ContainerStats {
+	return stats.ContainerStats{
+		Name:      name,
+		StartTime: metav1.NewTime(startTime.Add(-time.Minute)),
+	}
+}
+
+func genTestSummaryTerminatedContainerBlankStats(name string) stats.ContainerStats {
+	return stats.ContainerStats{
+		Name:      name,
+		StartTime: metav1.NewTime(startTime.Add(-time.Minute)),
+		CPU:       genTestSummaryBlankCPU(),
+		Memory:    genTestSummaryBlankMemory(),
+	}
+}
+
 func genTestSummaryContainer(name string, seed int) stats.ContainerStats {
 	return stats.ContainerStats{
 		Name:      name,
@@ -386,6 +432,12 @@ func genTestSummaryCPU(seed int) *stats.CPUStats {
 	return &cpu
 }
 
+func genTestSummaryBlankCPU() *stats.CPUStats {
+	return &stats.CPUStats{
+		Time: metav1.NewTime(scrapeTime),
+	}
+}
+
 func genTestSummaryZeroMemory(seed int) *stats.MemoryStats {
 	return &stats.MemoryStats{
 		Time:            metav1.NewTime(scrapeTime),
@@ -405,6 +457,12 @@ func genTestSummaryMemory(seed int) *stats.MemoryStats {
 		RSSBytes:        uint64Val(seed, offsetMemRSSBytes),
 		PageFaults:      uint64Val(seed, offsetMemPageFaults),
 		MajorPageFaults: uint64Val(seed, offsetMemMajorPageFaults),
+	}
+}
+
+func genTestSummaryBlankMemory() *stats.MemoryStats {
+	return &stats.MemoryStats{
+		Time: metav1.NewTime(scrapeTime),
 	}
 }
 

